plugins/@grouparoo/mailchimp/src/lib/export/exportProfile.ts

```diff
@@ -41,6 +42,11 @@ export const exportProfile: ExportProfilePluginMethod = async ({
     const oldMailchimpId = generateMailchimpId(
       oldProfileProperties["email_address"]
     );
+    if (!syncOperations.update) {
+      throw new Errors.InfoError(
+        "Destination sync mode does not allow updating existing profiles."
+      );
+    }
     try {
       // this will update the email of existing member only if the new email (email_address) do not exists yet.
       await client.put(`/lists/${listId}/members/${oldMailchimpId}`, {
```

Comment:
I'd like to understand how this behaves when OLD email does not exist. For example, if the old email is `email1@demo.com` and new email is `email2@demo.com`... What does the api return if `oldMailchimpId` does not exist in the destination? Will it error/404?

Reply:
nope, it returns 200 since the method allows you to add or update a list member.

Reply:
In that case, I think there may be some interesting behavior going on with email changes + `toDelete`.

- If the old email and new email do not exist, someone would be created by this PUT request and then immediately deleted by the updateProfile call. This should probably just not do anything instead of creating/deleting someone.
- If the old email does not exist but the new email does exist, I'm not sure what would happen. I'm assuming this would fail because the new email already exists, and then the code in the `catch` block would try to clear groups on the `oldMailchimpId`, which is nonexistent and could cause an error, not sure what the api returns.

Could you add tests for these cases with both `toDelete: true` and `toDelete: false`?